pkg/controllers/node/termination/controller.go

```diff
@@ -104,12 +120,24 @@ func (c *Controller) finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 		}
 		return reconcile.Result{RequeueAfter: 1 * time.Second}, nil
 	}
+	// In order for stateful pods to smoothly migrate from the terminating Node, we wait for VolumeAttachments
+	// to be cleaned up before terminating the node and removing it from the cluster.
+	if err := c.waitForVolumeDetachments(ctx, node, volumeDetachmentsBackoff); err != nil {
+		c.recorder.Publish(terminatorevents.NodeVolumeAttachmentsRemaining(node, err))
```

Comment:
Do we prefer logging a warning and resolving errors in waitForVolumeDetachments instead of publishing an event?